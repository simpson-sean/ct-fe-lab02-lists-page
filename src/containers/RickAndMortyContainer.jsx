import React, {Component } from 'react';
import { findCharacters } from '../services/RickAndMortyApi.js';
import CharacterList from '../components/app/characters/CharacterList.jsx';

export default class RickAndMortyContainer extends Component {
    state = {
        loading: true,
        characters: [],
    };

    componentDidMount() {
        findCharacters().then((characters) => 
          this.setState({ characters, loading: false })
        );
    }


    render() {
        const { loading, characters } = this.state;

        if (loading) {
          return (
            <img
                src="https://powerusers.microsoft.com/t5/image/serverpage/image-id/118082i204C32E01666789C?v=v2"
                alt="loading spinner"
            />    
        );
    }

    return <CharacterList characters={characters} />;
    } 
}