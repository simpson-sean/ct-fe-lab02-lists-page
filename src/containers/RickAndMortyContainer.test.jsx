import dotenv from 'dotenv';
dotenv.config();

import React from 'react';
import { render, screen } from '@testing-library/react';
import RickAndMortyContainer from './RickAndMortyContainer';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';

const server = setupServer(
    rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
        return res(
            ctx.json({
                results: [
                    {
                        id: 1,
                        name: 'Rick Sanchez (From API)',
                        status: 'alive',
                        image: 'example.com/image.png',
                    },                
                ],
            })
        );
    })
);

describe('RickAndMortyContainer', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('renders a list of characters', async () => {
        render(
        <MemoryRouter>
            <RickAndMortyContainer />
        </MemoryRouter>
        );
        screen.getByAltText('loading spinner')

        const ul = await screen.findByRole('list', { name: 'characters' }, {timeout: 3000});

        expect(ul).toMatchSnapshot();
    })
}
)