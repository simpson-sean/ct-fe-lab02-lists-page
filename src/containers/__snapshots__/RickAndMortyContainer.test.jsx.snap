// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyContainer renders a list of characters 1`] = `
<ul
  aria-label="characters"
>
  <div>
    <li>
      <a
        href="/1"
      >
        <img
          alt="Rick Sanchez (From API)"
          src="example.com/image.png"
        />
      </a>
    </li>
  </div>
</ul>
`;
